### DISCUSSION TITLE
Wireguard VPN traffic blocked by default

### DISCUSSION THREAD

**Discussion by zod076** [celzero/rethink-app#1811] [Q&A]
Hi, using rdns 0.5.5n on Android 14, with Wireguard set as the proxy, everything seems to be blocked. Nothing gets out. Turning off and deleting the Wireguard profile allows me to browse the web again.

The same Wireguard config when used with the official Wireguard app, everything works. What could be the issue here ?

I have rdns in DNS+Firewall mode(recommended)

**Comment by onetimecontributor** [comment]
as a matter of fact, Android 14 + rdns + wireguard (proton) works just fine for me.
so it's definitely not a generic issue.
I have set rdns as "always on" in Android's system settings, and my wireguard's profile is in both "lockdown" and "always on"

**Reply by zod076** [reply]
Anything else I could do to narrow down the issue ?

Like some other settings I missed ?

**Reply by zod076** [reply]
Ok, got a bit close. It looks like the issue is related to IPv6. My VPN has working IPv6 yet Rethink doesn't use it and after timing out it falls back to IPv4.

I tested this with one of those whatsmyipv6 test sites and only the VPN IPv4 address appears on the test page.

Is Rethink blocking IPv6 when Wireguard proxy is in use ?

**Reply by onetimecontributor** [reply]
> Is Rethink blocking IPv6 when Wireguard proxy is in use ?

sorry, can't test immediately, as I don't have relevant infra atm, though will have it soon 

disclaimer: I'm just a regular rdns user

**Reply by zod076** [reply]
No worries..

But that seems to be what I'm seeing as whatsmyip test sites only shows the VPN's IPv4 address.

**Comment by onetimecontributor** [comment]
just in case, have you set Configure - Network - Choose IP version to Auto ?

**Comment by zod076** [comment]
> just in case, have you set Configure - Network - Choose IP version to Auto ?

Yep, I think the VPN works but with IPv4. Meaning if the Wireguard Endpoint is IPv4 but the tunnel Addresses and DNS are IPv6 then rdns seems to block it somehow.

**Reply by ignoramous** [reply]
@zod076 Can you please see if things have improved in [v055t](https://github.com/celzero/rethink-app/releases/tag/v0.5.5t) (which is a pre-release). We haven't specifically tested for the scenario you are reporting here, but we've made a bunch of changes wrt WireGuard connectivity. May be v055t works, but if it doesn't, I'll open a GitHub issue to track this bug and fix it on priority one of the upcoming releases.

**Reply by zod076** [reply]
Been testing it for 45minutes now. My Wireguard VPN profile is dual-stack in _Simple_ mode.. Currently the status says _protected by wireguard_ and the _Proxy_ status fluctuates between _Active_ to _Failed_ .. when I tried to troubleshoot it, I notice it failed on IPv6 (https://ip6.me | https://ipv6only.me)..

When IPv6 requests fails, the _Proxy_ status is shown as _Active_, I'm guessing it's falling back to IPv4..so it's back to the earlier issue I had, which is the Wireguard VPN proxy only works with IPv4 consistently and often fails with a dual-stack VPN..

I have _Choose IP version_ set to _Auto_ and doing _Perform connectivity checks_ , _httpsv6_ is failing..

Edit 1: When I connect to my home VPN, browsing my LAN over IPv6 fails while over IPv4 it works.

**Reply by ignoramous** [reply]
Thanks for taking me up on testing v055t (:

> have Choose IP version set to Auto and doing Perform connectivity checks , httpsv6 is failing..

Rethink doesn't add routes for v6 if the underlying network won't route v6 (as noted by httpsv6 checks failing). That is, for dual stack WireGuard to work, the underlying network (ex: wifi / mobile) must also be dual stack.

You can scroll to the "VPN Stats" section in *About* -> *Stats* to see if the underlying network has v4/v6 routes (that is, if it is dual stack). If you can't understand what the output says, you can also paste the relevant bits from "VPN Stats" here and I'll take a look.

This is not a WireGuard limitation per se, but a limitation that arises from Rethink supporting running multiple WireGuard at once.
